src/util.h

```diff
@@ -132,15 +132,12 @@ static const secp256k1_callback default_error_callback = {
 } while(0)
 #endif
 
-/* Like assert(), but when VERIFY is defined, and side-effect safe. */
-#if defined(COVERAGE)
-#define VERIFY_CHECK(check)
-#define VERIFY_SETUP(stmt)
-#elif defined(VERIFY)
+/* Like assert(), but when VERIFY is defined. */
+#if defined(VERIFY)
 #define VERIFY_CHECK CHECK
 #define VERIFY_SETUP(stmt) do { stmt; } while(0)
 #else
-#define VERIFY_CHECK(cond) do { (void)(cond); } while(0)
+#define VERIFY_CHECK(cond)
 #define VERIFY_SETUP(stmt)
```

Comment:
c723b85: what does `VERIFY_SETUP` do? it looks really similar to `VERIFY_CHECK` and used in only 2 places in the codebase.

Reply:
The only two uses trace back to 72ae443afb536221a8523646ea4a88162341c3df where the macro was introduced.

My thinking is that the purpose of `VERIFY_SETUP` is to wrap code whose only purpose is to perform side effects ("setup") for further `VERIFY_CHECK`s. But assuming this interpretation, I don't really understand 72ae443afb536221a8523646ea4a88162341c3df. 

The effect of the two existing `VERIFY_SETUP` uses is that they initialize `r->x` and `r->y` but that's done anyway in the current code, immediately afterward: 
https://github.com/bitcoin-core/secp256k1/blob/5814d8485cbb074115c03b587b6799a812e0f267/src/ecmult_const_impl.h#L90-L95

@sipa Is there a reason why you kept these in #1184?

Anyway, I suggest:
 - Drop the two uses of `VERIFY_SETUP`.
 - Drop `VERIFY_SETUP`. (Our implicit convention for such code is to nest it in an `#ifdef VERIFY` block.)